sites/funmooc/src/backend/base/tests/test_cache.py

```diff
@@ -0,0 +1,109 @@
+    """
+    Test suite for the RedisCacheWithFallback
+
+    Credits:
+    - https://github.com/Kub-AT/django-cache-fallback
+    """
+
+    @override_settings(
+        CACHES={
+            "default": {
+                "BACKEND": "base.cache.RedisCacheWithFallback",
+                "LOCATION": "mymaster/redis-sentinel:26379,redis-sentinel:26379/0",
+                "OPTIONS": {
+                    "CLIENT_CLASS": "richie.apps.core.cache.SentinelClient",
+                },
+            },
+            "fallback": {
+                "BACKEND": "django.core.cache.backends.dummy.DummyCache",
+            },
+        }
+    )
+    def test_client(self):
+        """Test class instance of caches"""
+        client = RedisCacheWithFallback(None, {})
+        self.assertIs(type(client), RedisCacheWithFallback)
+        self.assertIs(type(client._redis_cache), RedisCache)
+        self.assertIs(type(client._fallback_cache), DummyCache)
+
+    @mock.patch.object(RedisCacheWithFallback, "_call_fallback_cache")
+    @mock.patch.object(RedisCacheWithFallback, "_call_redis_cache")
+    def test_get_redis_cache(self, redis_cache_mock, fallback_cache_mock):
+        """Test that redis_cache is used by default."""
+        client = RedisCacheWithFallback(None, {})
+        client.get("irrelevent")
+
+        redis_cache_mock.assert_called_once()
+        fallback_cache_mock.assert_not_called()
+
+    @mock.patch("base.cache.logger")
+    @mock.patch.object(RedisCacheWithFallback, "_call_fallback_cache")
+    @mock.patch.object(RedisCacheWithFallback, "_call_redis_cache")
+    def test_get_fallback_cache(
+        self, redis_cache_mock, fallback_cache_mock, logger_mock
+    ):
+        """
+        Test case when redis_cache raises an exception,
+        logger logs the exception then fallback_cache takes over
+        """
+        client = RedisCacheWithFallback(None, {})
+        client.get("irrelevent")
+
+        redis_cache_mock.assert_called_once()
+        fallback_cache_mock.assert_not_called()
+
+        redis_cache_mock.side_effect = Exception()
+        client.get("irrelevent")
+
+        logger_mock.warning.assert_called_with(
+            "[DEGRADED CACHE MODE] - Switch to fallback cache"
+        )
+        logger_mock.exception.assert_called_once()
+        fallback_cache_mock.assert_called_once()
+
+    @override_settings(
+        CACHES={
+            "fallback": {
+                "BACKEND": "django.core.cache.backends.dummy.DummyCache",
+            },
+        }
+    )
+    @mock.patch.object(RedisCacheWithFallback, "_call_redis_cache")
+    def test_invalidate_fallback_cache(self, redis_cache_mock):
+        """
+        Test that fallback_cache is invalidate each 60 seconds
+        when Redis is up
+        """
+        m = mock.MagicMock()
+        m.delete = mock.Mock(return_value="delete")
+
+        with mock.patch(
+            "django.core.cache.backends.locmem.LocMemCache", return_value=m
+        ) as locmem_mock:
```

Comment:
I suppose you tried this :crossed_fingers:
```suggestion
        from django.core.cache.backends.locmem import LocMemCache
        with mock.patch.object(LocMemCache, "delete", return_value="delete") as locmem_mock:
```

Reply:
Yes, `call_count` is still 0 😕

Reply:
`locmem_mock.delete.call_count` was 0, `locmem_mock.call_count` is right ...